api/v1alpha1/bgppeer_webhook.go

```diff
@@ -0,0 +1,182 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging bgppeer-webhook
+var bgppeerlog = logf.Log.WithName("bgppeer-webhook")
+
+func (bgpPeer *BGPPeer) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(bgpPeer).
+		Complete()
+}
+
+const (
+	BGPDefaultPort     = 179
+	BGPDefaultHoldTime = 90 * time.Second
+)
+
+var _ webhook.Defaulter = &BGPPeer{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (bgpPeer *BGPPeer) Default() {
```

Comment:
Can we achieve this with defaults in the CRD? See https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting